contracts/v2/PolygonZkEVMBridgeV2.sol

```diff
@@ -726,6 +735,34 @@ contract PolygonZkEVMBridgeV2 is
             ];
     }
 
+    /**
+     * @notice Set the address of a wrapper using the token information if already exist
+     * @dev This function is used to allow any existing token to be mapped with
+     *      origin token. Wrapper contract should handle mint/burn of the existing token.
+     * @param originNetwork Origin network
+     * @param originTokenAddress Origin token address, 0 address is reserved for ether
+     * @param wrappedTokenAddress Arbitary contract that implements TokenWrapped interface
+     */
+    function setCustomTokenMapping(
+        uint32 originNetwork,
+        address originTokenAddress,
+        address wrappedTokenAddress,
+        address existingTokenAddress
+    ) external onlyRollupManager {
+        // Handle claimAsset on target chain
+        bytes32 tokenInfoHash = keccak256(
+            abi.encodePacked(originNetwork, originTokenAddress)
+        );
+        tokenInfoToWrappedToken[tokenInfoHash] = wrappedTokenAddress;
+
+        // Handle bridgeAsset from origin chain
+        wrappedTokenToTokenInfo[existingTokenAddress] = TokenInformation(
+            originNetwork,
+            originTokenAddress
+        );
+        existingTokenToWrapper[existingTokenAddress] = wrappedTokenAddress;
```

Comment:
suggestion: add a comment noting that if `setCustomTokenMapping` is called multiple times for the same `existingTokenAddress`, this will override the previous calls and only keep the last `wrappedTokenAddress`